### PR TITLE
Hanabi improvements

### PR DESCRIPTION
**Pull request by andrejak**
- show discard pile in sorted order
- add Royal Favor variant (more like what we typically play - playing until all piles are 5, and until players run out of cards)
- show whose turn it is
- selection on other players hands should move now as well
- hide "Give info" button when no info tokens left
- fix end condition (play ended one step too early before)

- refactor code a bit 
- added some tests
- slightly prettier design for the UI

### CODE REVIEW

**Comment by andrejak** [comment]
@tdawes if you don't mind, I might merge this so I can test it with a couple friends :) (and then I'll add any more restyling or fixes in another pr)

**Comment by tdawes** [comment]
Hey - sorry I didn't even notice this was open. Feel free to merge!

File: games/hanabi/src/api/index.ts [hunk 1]
```diff
@@ -17,6 +19,10 @@ export type Colour = typeof colours[number];
 export const numbers = [1, 2, 3, 4, 5] as const;
 export type Number = typeof numbers[number];
 export const maxCardNum = numbers[numbers.length - 1];
+export const getColours = (gameType: GameType): Colour[] =>
+  gameType === "rainbow"
+    ? colours.slice()
```
Comment (tdawes) [inline games/hanabi/src/api/index.ts]: Why are you cloning?

Comment (andrejak) [reply games/hanabi/src/api/index.ts]: oh, this was a weird hack to get around a type error that I then forgot about - will look into a better fix (sorry, clicked merge a few minutes ago, so will address any comments in a new pr)

**Pull request merged**